src/Randomizer.Data/Services/MetadataService.cs

```diff
@@ -211,7 +211,7 @@ public RoomInfo Room(Type type)
         /// <returns>
         /// A new <see cref="RoomInfo"/> for the specified room.
         /// </returns>
-        public RoomInfo Room(Room room)
+        public RoomInfo? Room(Room room)
```

Comment:
Looks like this introduced some possible null reference warnings. Can you see about addressing those?